packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -171,7 +223,64 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
     parentRef.current.focus();
   };
 
+  setSelectedItemIds = (e: React.SyntheticEvent, selectedItemIds: string[]) => {
+    _.invoke(this.props, 'onSelectedItemIdsChange', e, { ...this.props, selectedItemIds });
+
+    this.setState({
+      selectedItemIds,
+    });
+  };
+
+  processItemsForSelection = (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+    let { selectedItemIds } = this.state;
+    const { id, selectableParent, items, expanded } = treeItemProps;
+    const treeItemHasSubtree = hasSubtree(treeItemProps);
+    const isExpandedSelectableParent = treeItemHasSubtree && selectableParent && expanded;
+
+    // parent must be selectable and expanded in order to procced with selection, otherwise return
+    if (treeItemHasSubtree && !(selectableParent && expanded)) {
+      return;
+    }
+
+    // if the target is equal to currentTarget it means treeItem should be collapsed, not procced with selection
+    if (isExpandedSelectableParent && e.target === e.currentTarget) {
+      return;
+    }
+
+    // push all tree items under particular parent into selection array
+    // not parent itself, therefore not procced with selection
+    if (isExpandedSelectableParent) {
```

Comment:
As agreed with @jurokapsiar - select all should also select nested children

Reply:
Solved :)